Решение заданий по Sprint 5. Многопоточность, Kotlin WebServer, Spring Framework

Nihrom77 opened the pull request:
Решение задания по webserver

Commit:
Решение задания по созданию сервера для обработки HTTP-запросов и возвращению данных из виртуальной файловой системы.

shariktlt-devops reviewed (changes requested):
:no_entry: build failed with modules spring-mvc, see: https://jenkins.shariktlt.ru/job/KotlinSchool/job/PR_Builder/479/

Commit:
Добавил правки в xml-конфигурацию, чтобы проходили тесты.

shariktlt-devops reviewed (changes requested):
:no_entry: build failed with modules spring-mvc, see: https://jenkins.shariktlt.ru/job/KotlinSchool/job/PR_Builder/487/

Commit:
Добавил логику по определению бинов в конфигурации для прохождения тестов.

Commit:
Добавил сервисы singleton и prototype для прохождения тестов.

shariktlt-devops reviewed (changes requested):
:no_entry: build failed with modules spring-mvc, see: https://jenkins.shariktlt.ru/job/KotlinSchool/job/PR_Builder/488/

Commit:
Добавил логику для инициализации бинов, чтобы проходили тесты.

Commit:
Добавил правки для условий выбора бинов, чтобы проходили тесты.

shariktlt-devops reviewed (changes requested):
:no_entry: build failed with modules spring-mvc, see: https://jenkins.shariktlt.ru/job/KotlinSchool/job/PR_Builder/493/

Commit:
Решение задания для CRUD-сервиса на примере адресной книги.

shariktlt-devops reviewed (approved):
:ok: build ok, see: https://jenkins.shariktlt.ru/job/KotlinSchool/job/PR_Builder/531/

SharikTLT reviewed (commented):
spring mvc выполнено без замечаний

SberKotlinSchool commented on spring-java-config/src/main/kotlin/ru/sber/config/AnotherServicesConfig.kt:
```diff
@@ -1,8 +1,22 @@
+
+    @Primary
+    @Bean
+    fun fourthService(): Any {
```
Вместо реализации метода, возвращающего бин, я бы рекомендовал обратить внимание на аннотацию @ComponentScan и в каком пакете она ищет нужные бины.

Nihrom77 replied:
Поправил параметр аннотации ComponentScan для исключения бина fourthService из неактуального конфига.

SberKotlinSchool commented on spring-dependency-injection/src/main/kotlin/ru/sber/services/SeveralServices.kt:
```diff
@@ -23,12 +23,13 @@ class SecondServiceImpl : ServiceInterface {
 
     override fun toString(): String {
-        return "SeveralBeanInjectionService(services=$services)"
+        return "SeveralBeanInjectionService(services=${listOf(firstService, secondService)})"
```
Неплохая попытка, но давай лучше обратим внимание на тип переменной services)
Может это наведет на мысль о том, почему сейчас вывод в тесте не соответствует тому, что ожидаем получить.

Nihrom77 replied:
Поменял тип коллекции для сбора бинов типа ServiceInterface.

Commit:
Изменил логику определения бина "fourthService" через изменения параметра сканирования пакетов для контекста.

Commit:
Изменил логику поиска бинов типа ServiceInterface в контексте.
Поскольку существует бин с именем "services": ArrayList<ServiceInterface> уже существует, то можно собрать бины типа ServiceInterface в другой тип коллекции.

shariktlt-devops reviewed (approved):
:ok: build ok, see: https://jenkins.shariktlt.ru/job/KotlinSchool/job/PR_Builder/580/
